src/Microsoft.OData.Core/JsonLight/ReorderingJsonReader.cs

```diff
@@ -31,6 +31,38 @@ internal ReorderingJsonReader(IJsonReader innerReader, int maxInnerErrorDepth)
             Debug.Assert(innerReader != null, "innerReader != null");
         }
 
+        /// <summary>
+        /// Creates a stream for reading a stream value.
+        /// </summary>
+        /// <returns>A Stream used to read a stream value.</returns>
+        public override Stream CreateReadStream()
+        {
+            Stream result = new MemoryStream(this.Value == null ? new byte[0] :
+                Convert.FromBase64String((string)this.Value));

```

Comment:
>FromBase64String((string)this.Value)); [](start = 24, length = 38)

how about if the "value" can't convert from Base64 string?

Reply:
Added a binary format exception for attempting to read a non-Base64 stream as a stream.

---
In reply to: [266668885](https://github.com/OData/odata.net/pull/1204#discussion_r266668885) [](ancestors = 266668885)